searchcore/src/vespa/searchcore/proton/matchengine/matchengine.cpp

```diff
@@ -141,11 +141,11 @@ MatchEngine::performSearch(search::engine::SearchRequest::Source req,
     ret->request = req.release();
     ret->setDistributionKey(_distributionKey);
     if (ret->request->trace().getLevel() > 0) {
-        ret->request->trace().getRoot().setLong("distribution-key", _distributionKey);
+        ret->request->trace().getRoot()->setLong("distribution-key", _distributionKey);
         ret->request->trace().done();
         search::fef::Properties & trace = ret->propertiesMap.lookupCreate("trace");
         vespalib::SmartBuffer output(4096);
-        vespalib::slime::BinaryFormat::encode(ret->request->trace().getSlime(), output);
+        vespalib::slime::BinaryFormat::encode(*ret->request->trace().getSlime(), output);
```

Comment:
getSlime should lazily create the slime object to avoid nullptr crash. Combine with additional has_slime predicate to add conditional behavior. Also consider returning by reference instead of by pointer.

Reply:
Fixed as discussed